File: src/main/java/com/Backend/music_app/dto/request/ArtistCreateRequest.java
package com.Backend.music_app.dto.request;
import com.Backend.music_app.dto.response.Item.AlbumItemResponse;
import com.Backend.music_app.dto.response.Item.SongItemResponse;
import com.fasterxml.jackson.annotation.JsonInclude;
import lombok.*;
import lombok.experimental.FieldDefaults;

import java.util.List;

@JsonInclude(JsonInclude.Include.NON_NULL)
@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
@FieldDefaults(level = AccessLevel.PRIVATE)
public class ArtistCreateRequest {
    String name;
    String avatarUrl;
    String bio;
    String country;
    List<SongItemResponse> topSongs;
    List<AlbumItemResponse> albums;
}

File: lib/data/createRequest/ArtistCreate_request.dart
import 'package:sq_mp3/data/model/Album_model.dart';
import 'package:sq_mp3/data/model/Song_item_model.dart';

class ArtistCreateRequest {
  final String name;
  final String bio;
  final String avatarUrl;
  final String country;
  List<SongItemModel> topSongs;
  List<AlbumModel> albums;

  ArtistCreateRequest({
    required this.name,
    required this.bio,
    required this.avatarUrl,
    required this.country,
    required this.topSongs,
    required this.albums,
  });

  Map<String, dynamic> toJson() {
    return {
      'name': name,
      'bio': bio,
      'avatarUrl': avatarUrl,
      'country': country,
      'topSongs': topSongs.map((x) => x.toJson()).toList(),
      'albums': albums.map((x) => x.toJson()).toList(),
    };
  }
}
